packages/gator-permissions-snap/src/clients/types.ts

```diff
@@ -96,6 +96,37 @@ export type TokenBalanceAndMetadata = {
   iconUrl?: string;
 };
 
+/**
+ * Represents a transaction receipt from the blockchain.
+ * As defined in the Ethereum JSON-RPC API(https://docs.metamask.io/services/reference/zksync/json-rpc-methods/eth_gettransactionreceipt/)
+ */
+export type TransactionReceipt = {
+  blockHash: Hex;
+  blockNumber: Hex;
+  contractAddress: Hex | null;
+  cumulativeGasUsed: Hex;
+  effectiveGasPrice: Hex;
+  from: Hex;
+  gasUsed: Hex;
+  logs: {
+    address: Hex;
+    blockHash: Hex;
+    blockNumber: Hex;
+    data: Hex;
+    logIndex: Hex;
+    removed: boolean;
+    topics: Hex[];
+    transactionHash: Hex;
+    transactionIndex: Hex;
+  }[];
+  logsBloom: Hex;
+  status: Hex;
+  to: Hex;
```

Comment:
Contract creation transactions have null `to` address:
```suggestion
  to: Hex | null;
```

But as suggested below, we could define this as a narrowed zod type. Then when we validate, it will (by default) discard anything not defined in the type, so we can ignore all fields we don't need.